apiserver/facades/client/charmhub/convert.go

```diff
@@ -142,32 +146,44 @@ func transformInfoChannelMap(channelMap []transport.InfoChannelMap) ([]string, m
 	return trackList, channels
 }
 
-func convertPlatforms(in []transport.Platform) []params.Platform {
+// convertBasesToPlatforms converts a base to a platform.  Is mean to be a short
+// term solution due to schedule.  It should be removed once platforms are
+// replaced with bases through out the code.
+func convertBasesToPlatforms(in []transport.Base, isKub bool) []params.Platform {
 	out := make([]params.Platform, len(in))
 	for i, v := range in {
+		channel := v.Channel
+		if isKub {
+			// Kubernetes is not a valid series for a base.  Instead use the latest
+			// LTS version of ubuntu.
+			channel = coreseries.LatestLts()
+		}
+		series, _ := coreseries.VersionSeries(channel)
 		out[i] = params.Platform{
 			Architecture: v.Architecture,
-			OS:           v.OS,
-			Series:       v.Series,
+			OS:           v.Name,
+			Series:       series,
 		}
 	}
 	return out
 }
 
-func convertCharm(info transport.InfoResponse) (*params.CharmHubCharm, []string) {
+func convertCharm(info transport.InfoResponse) (*params.CharmHubCharm, []string, bool) {
```

Comment:
We will need to detect whether we are dealing with V2 metadata here and determine the series from bases in manifest.yaml. Not to what extent this is mandatory for 2.9.0.

Reply:
I'll add a TODO to isKubernetes()